File: MovieCatalogQw/app/src/main/java/com/unlistedi/moviecatalogqw/ui/movies/MoviesAdapter.java
package com.unlistedi.moviecatalogqw.ui.movies;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.unlistedi.moviecatalogqw.DetailActivity;
import com.unlistedi.moviecatalogqw.R;
import com.unlistedi.moviecatalogqw.data.MovieEntity;

import java.util.ArrayList;

public class MoviesAdapter extends RecyclerView.Adapter<MoviesAdapter.MoviesViewHolder> {
    private ArrayList<MovieEntity> movies;

    public MoviesAdapter(ArrayList<MovieEntity> movies) {
        this.movies = movies;
    }

    @NonNull
    @Override
    public MoviesViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new MoviesViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_movie, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull final MoviesViewHolder holder, final int position) {
        holder.tvListTitle.setText(movies.get(position).getTitle());
        holder.tvListScore.setText(String.valueOf(movies.get(position).getScore()));

        Glide.with(holder.itemView.getContext())
                .load(movies.get(position).getImg_path())
                .into(holder.ivPoster);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(holder.itemView.getContext(), DetailActivity.class);
                intent.putExtra(DetailActivity.EXTRA_MOVIE, movies.get(position));
                holder.itemView.getContext().startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return this.movies.size();
    }

    public void setMovies(ArrayList<MovieEntity> movies) {
        this.movies = movies;
    }

    public class MoviesViewHolder extends RecyclerView.ViewHolder{
        TextView tvListTitle, tvListScore;
        ImageView ivPoster;

        public MoviesViewHolder(@NonNull View itemView) {
            super(itemView);
            tvListTitle = itemView.findViewById(R.id.tvListTitle);
            tvListScore = itemView.findViewById(R.id.tvListScore);
            ivPoster = itemView.findViewById(R.id.ivPoster);
        }
    }
}

File: MovieCatalogueKu/app/src/main/java/com/unlistedi/moviecatalogueku/ui/movies/MoviesAdapter.kt
package com.unlistedi.moviecatalogueku.ui.movies

import android.content.Intent
import android.os.Build
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.RequiresApi
import androidx.recyclerview.widget.RecyclerView
import com.unlistedi.moviecatalogueku.DetailActivity
import com.unlistedi.moviecatalogueku.R
import com.unlistedi.moviecatalogueku.data.MovieEntity
import com.unlistedi.moviecatalogueku.utils.GlideApp

class MoviesAdapter(private var movies : ArrayList<MovieEntity>) : RecyclerView.Adapter<MoviesAdapter.MoviesViewHolder>(){

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MoviesViewHolder {
        return MoviesViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.items_movie, parent,false))
    }

    override fun getItemCount(): Int {
        return movies.size
    }

    @RequiresApi(Build.VERSION_CODES.O)
    override fun onBindViewHolder(holder: MoviesViewHolder, position: Int) {
        holder.tvListTitle.text = movies[position].title
        holder.tvListScore.text = movies[position].score.toString()

        GlideApp.with(holder.itemView.context)
            .load(movies[position].img_path)
            .into(holder.ivPoster)

        holder.itemView.setOnClickListener {
            val intent = Intent(holder.itemView.context, DetailActivity::class.java)
            intent.putExtra(DetailActivity.EXTRA_MOVIE, movies[position])
            holder.itemView.context.startActivity(intent)
        }
    }

    class MoviesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
        val tvListTitle : TextView = itemView.findViewById(R.id.tvListTitle)
        val tvListScore : TextView = itemView.findViewById(R.id.tvListScore)
        val ivPoster : ImageView = itemView.findViewById(R.id.ivPoster)
    }

}